src/commands/report.ts

```diff
@@ -30,185 +33,230 @@ export const data = new SlashCommandBuilder()
+          ephemeral: true,
+        });
+        return;
+      }
 
-  const title = submissionInteraction.fields.getTextInputValue("titleInput");
-  const description = submissionInteraction.fields.getTextInputValue("descriptionInput");
-  const contactInfo = submissionInteraction.fields.getTextInputValue("contactInput");
+      // If no existing reports, continue with the original flow
+    }
+  } catch (error) {
+    logger.error(error, "Unable to check asset status (url=%s)", urlInput);
+  }
 
-  // Getting the Discord user information
-  const discordAvatarUrl = user.displayAvatarURL();
-  const discordPublicUsername = user.username;
-  const discordFormattedUsername = `${user.username}#${user.discriminator}`; // username in "user#1234" format
-  const externalUser = {
-    platform: "discord",
-    platformIdentifier: user.id,
-    avatarUrl: discordAvatarUrl,
-    displayName: discordFormattedUsername,
-  };
+  // Show the modal to the user
+  const modal = generateModal(user, options, guildId);
+  await interaction.showModal(modal);
 
+  // Wait for modal submission
   try {
-    const response = await chainpatrol.report.create({
-      discordGuildId: guildId ?? undefined,
-      externalReporter: externalUser,
-      title,
-      description,
-      contactInfo,
-      assets: [{ content: url, status: "BLOCKED" }],
+    const submissionInteraction = await interaction.awaitModalSubmit({
+      filter: (i) => i.customId === "reportModal" && i.user.id === user.id,
+      time: 5 * 60 * 1000, // 5 minutes
     });
 
-    await submissionInteraction.reply({
-      content: `✅ Thanks for submitting a report for \`${escapedUrl}\` ! \n\nWe've sent this report to the **${response.organization.name}** team and **ChainPatrol** to conduct a review. Once approved the report will be sent out to wallets to block.\n\nThanks for doing your part in making this space safer 🚀`,
+    const url = submissionInteraction.fields.getTextInputValue("urlInput");
+    const escapedUrl = defangUrl(url);
+    const title = submissionInteraction.fields.getTextInputValue("titleInput");
+    const description =
+      submissionInteraction.fields.getTextInputValue("descriptionInput");
+    const contactInfo = submissionInteraction.fields.getTextInputValue("contactInput");
+
+    // Create buttons
+    const editButton = new ButtonBuilder()
+      .setCustomId("edit_report")
+      .setLabel("Edit")
+      .setStyle(ButtonStyle.Secondary);
+
+    const submitButton = new ButtonBuilder()
+      .setCustomId("submit_report")
+      .setLabel("Submit")
+      .setStyle(ButtonStyle.Primary);
+
+    const row = new ActionRowBuilder<ButtonBuilder>().addComponents(
+      editButton,
+      submitButton,
+    );
+
+    // Show ephemeral message with content from the modal and buttons
+    const reply = await submissionInteraction.reply({
+      content: `Please review your report for \`${escapedUrl}\`:
```

Comment:
Try to follow the copy and structure from the in-app report flow:

![CleanShot 2024-09-12 at 10 32 37@2x](https://github.com/user-attachments/assets/a3246df0-ff87-4c9e-ae87-4d4d7b55855f)